### DISCUSSION TITLE
Feature request: rdFMCS timeout support values <1

### DISCUSSION THREAD

**Discussion by and-tos** [rdkit/rdkit#5094] [Ideas]
When calculating the MCS between two molecules, a very short timeout can make sense. Currently, the argument has to be an int. Is there a reason for this?

```
>>> from rdkit import Chem
>>> from rdkit.Chem import rdFMCS
>>> m1 = Chem.MolFromSmiles('c1ccccc1')
>>> m2 = Chem.MolFromSmiles('c1ccccc(O)1')
>>> rdFMCS.FindMCS([m1, m2], completeRingsOnly=True, timeout=1)
<rdkit.Chem.rdFMCS.MCSResult object at 0x14b0ffcb3fa0>
>>> rdFMCS.FindMCS([m1, m2], completeRingsOnly=True, timeout=0.5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Boost.Python.ArgumentError: Python argument types in
    rdkit.Chem.rdFMCS.FindMCS(list)
did not match C++ signature:
    FindMCS(boost::python::api::object mols, RDKit::PyMCSParameters {lvalue} parameters)
    FindMCS(boost::python::api::object mols, bool maximizeBonds=True, double threshold=1.0, unsigned int timeout=3600, bool verbose=False, bool matchValences=False,ngMatchesRingOnly=False, bool completeRingsOnly=False, bool matchChiralTag=False, RDKit::AtomComparator atomCompare=rdkit.Chem.rdFMCS.AtomCompare.CompareElements, RndComparator bondCompare=rdkit.Chem.rdFMCS.BondCompare.CompareOrder, RDKit::RingComparator ringCompare=rdkit.Chem.rdFMCS.RingCompare.IgnoreRingFusion, std::__cxx11:tring<char, std::char_traits<char>, std::allocator<char> > seedSmarts='')

```

**Comment by IchiruTake** [comment]
According to the .cpp module they declared on `Code/GraphMol/FMCS.cpp` and `Code/GraphMol/FMCS.h`, they initialized `timeout` attribute to be `unsigned` and set default to be -1 (which by then converted to 1. The main reason they don't accept double or float may came from function `MCSProgressCallbackTimeout()` where they used unsigned long long for nanosecond time measurement. 

I am not confident at C++ that if it can automatically convert to double or not, but if not, incorrect signature is what you get as above.

**Reply by IchiruTake** [reply]
If they have intention to fix, they can declare the value to be double and do some casting from int to float/double. And probably do some argument constraint to enforce postive value MCSParameters.timeout. By then when checking the result, they can get the integer part of `MCSParameters.timeout * 1e9`. Although this idea can help you get the idea but it may cause a minor loss in time measurement. However, I don't think this loss is significant and can be acceptable. However, unless user can intentionally initialize the value of `timeout` to be too small, let say 1e-10, they may set another constraint such as `timeout` must be greater than 1ms (1e-3 sec)
